test/fixtures/simple-app/boot/custom.customjs

```diff
@@ -0,0 +1,8 @@
+process.bootFlags.push('customLoaded');
+module.exports = function(app, callback) {
+  process.bootFlags.push('customStarted');
+  process.nextTick(function() {
+    process.bootFlags.push('customFinished');
+    callback();
+  });
```

Comment:
This is unnecessarily complex. The thing to verify in your test is whether loopback-boot recognises a custom extension. There is no need to verify how boot script are loaded/executed again.

```js
'use strict';

module.exports = function(app, callback) {
  process.bootFlags.push('customjs');
  callback();
};
```

Same comment applies to `custom.customjs2`.